### DISCUSSION TITLE
Buffer output between sessions under split

### DISCUSSION THREAD

**Discussion by miguelalbuquerque** [scandum/tintin#96] [General]
Greetings,

I've been trying to solve a limitation without success. I've run out of ideas apart from asking here!

Basically the default behaviour of starting on the default gts session, starting a new one, and quitting (back to gts) keeps the output in the same window buffer. I can scroll up, etc.
However, when I'm using "split" it seems each session has its own independent buffer/console output.

in short: when a mud quits (idling out, or manually quitting), I want to still have the buffer of that recently quit session in my screen (now in gts). To see the messages the mud shows on quit, if someone messaged me something before I idled out, etc.

Is there any way to achieve this while using split? Using it is essential for me, as I've customised a lot of my mud output to show the map, current map room info, etc.

Thanks!

**Comment by scandum** [comment]
You could try:
```
#EVENT {SESSION DISCONNECTED}
{
    #buffer write last.log;
    #gts #line ignore #scan txt last.log
}
```
Or if you want to keep it internal:
```
#EVENT {SESSION DISCONNECTED}
{
    #buffer get buffer 1 -1;
    #loop 1 &buffer[] line
    {
        #gts #line ignore #showme {$buffer[+$line]}
    }
}
```

**Reply by miguelalbuquerque** [reply]
Thanks. I'm afraid it does not work. I've tried several combinations or splits, trying to unsplit before logging, etc..

Here's a demonstrating example (my real case does a split with a side box to show a map in, taking into account the current screen size):

```
#EVENT {SESSION DISCONNECTED}
{
    #unsplit; #NOP just in case, still doesn't work.
    #buffer get buffer 1 -1;
    #loop 1 &buffer[] line
    {
        #gts #line ignore #showme {$buffer[+$line]}
    }
}

#EVENT {SESSION CONNECTED}
{
    #split 1;
}

#session t2t t2tmud.org 9999;
```

**Comment by scandum** [comment]
Keep in mind you need to end the #NOP with a ;

You can hit page up/down to refresh the buffer, but the following should do so automatically:
```
#EVENT {SESSION DISCONNECTED}
{
    #buffer get buffer 1 -1;
    #loop 1 &buffer[] line
    {
        #gts #line ignore #showme {$buffer[+$line]}
    }
}

#EVENT {SESSION CONNECTED}
{
    #split 1;
}

#event {SESSION ACTIVATED}
{
        #buffer end
}
```

**Comment by miguelalbuquerque** [comment]
That worked!
Thank you.

Great client, and great support, as always! :)
